libtelemetry/src/full/java/com/mapbox/android/telemetry/UploadJobIntentService.java

```diff
@@ -0,0 +1,52 @@
+package com.mapbox.android.telemetry;
+
+import android.content.Context;
+import android.content.Intent;
+import android.support.annotation.NonNull;
+import android.support.v4.app.JobIntentService;
+import android.support.v4.content.LocalBroadcastManager;
+import android.util.Log;
+
+import com.google.gson.Gson;
+import com.mapbox.libupload.MapboxUploader;
+
+import java.io.IOException;
+import java.util.List;
+
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.Response;
+
+public class UploadJobIntentService extends JobIntentService implements Callback {
+  private static final String LOG_TAG = "UploadJobIntentService";
+  private static final int JOB_ID = 564321;
+  private static MapboxUploader.MapboxUploadClient<List<Event>, UploadJobIntentService> uploadClient;
+  private Gson gson = new Gson();
+
+  public void schedule(Context context, List<Event> events, MapboxUploader.MapboxUploadClient mapboxUploadClient) {
+    uploadClient = mapboxUploadClient;
+
+    Intent intent = new Intent("UploadJob");
```

Comment:
i wonder what's the reason behind fetching events prior to scheduling its upload?